examples/nodejs/token-vending-machine/infrastructure/lib/token-vending-machine-stack.ts

```diff
@@ -46,6 +48,48 @@ export class TokenVendingMachineStack extends cdk.Stack {
       requestTemplates: {'application/json': '{ "statusCode": "200" }'},
     });
 
-    api.root.addMethod('GET', tvmIntegration);
+    if (config.authenticationMethod === "lambda-authorizer") {
+      const authLambda = new lambdaNodejs.NodejsFunction(this, 'MomentoTokenVendingMachineAuthorizer', {
+        functionName: 'MomentoTokenVendingMachineAuthorizer',
+        runtime: lambda.Runtime.NODEJS_18_X,
+        entry: path.join(__dirname, '../../lambda/authorizer/authorizer.ts'),
+        projectRoot: path.join(__dirname, '../../lambda/authorizer'),
+        depsLockFilePath: path.join(__dirname, '../../lambda/authorizer/package-lock.json'),
+        handler: 'handler',
+        timeout: cdk.Duration.seconds(30),
+        memorySize: 128,
+      });
+
+      const authorizer = new apig.RequestAuthorizer(this, 'MomentoTokenVendingMachineTokenAuthorizer', {
+        handler: authLambda,
+        identitySources: [apig.IdentitySource.header('username'), apig.IdentitySource.header('password')],
+        resultsCacheTtl: cdk.Duration.seconds(5),
+      });
+
+      api.root.addMethod('GET', tvmIntegration, {
+        authorizer: authorizer,
+      });
+    }
+    else if (config.authenticationMethod === "amazon-cognito") {
+      const userPool = new cognito.UserPool(this, 'MomentoTokenVendingMachineUserPool', {
+        userPoolName: 'MomentoTokenVendingMachineUserPool',
+        signInAliases: {
+          username: true
+        },
+        passwordPolicy: {
+          minLength: 8,
+          requireSymbols: true,
+        }
+      });
```

Comment:
just out of curiosity, does CDK have a resource type for users in the pool here? i.e. since you are defining the pool in CDK, can we also add members to it in CDK?

Reply:
It might, I can take a deeper look into that today